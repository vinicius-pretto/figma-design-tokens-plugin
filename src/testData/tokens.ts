import TokenType from "../consts/TokenType";

const tokens = [
  {
    id: "99dd84d3-a355-4a3b-a061-6bab7a8bf285",
    type: TokenType.COLOR,
    name: "color-dark",
    value: "#222222",
  },
  {
    id: "757e0617-d68f-498c-af84-99579f0783d4",
    type: TokenType.COLOR,
    name: "color-yellow",
    value: "#ccc000",
  },
];

export default tokens;
