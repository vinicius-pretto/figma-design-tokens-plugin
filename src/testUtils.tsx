import * as React from "react";
import { render } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import configureStore from "./ui/redux/store";

const Providers = ({ children }) => {
  return <ReduxProvider store={configureStore()}>{children}</ReduxProvider>;
};

const customRender = (ui) => {
  return render(ui, {
    wrapper: Providers,
  });
};

export * from "@testing-library/react";
export { customRender as render };
