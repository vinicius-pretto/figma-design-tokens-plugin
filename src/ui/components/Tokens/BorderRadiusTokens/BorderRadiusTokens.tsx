import * as React from "react";
import _isEmpty from "lodash/isEmpty";
import BorderRadiusToken from "./BorderRadiusToken";
import TokenType from "../../../../consts/TokenType";
import Token from "../../../../consts/Token";

const BorderRadiusTokens = ({ tokens, onDelete }) => {
  const borderRadiusTokens = tokens.filter(
    (token: Token) => token.type === TokenType.BORDER_RADIUS
  );

  const onCreate = () => {};

  return (
    <section className="border-bottom p-sm">
      <div className="tokens-section-header">
        <h3 className="section-title">Border Radius</h3>

        <div
          className="icon-button"
          role="button"
          aria-label="Add token"
          tabIndex={0}
          onClick={onCreate}
          data-testid="add-token-btn"
        >
          <div className="icon icon--plus"></div>
        </div>
      </div>

      {_isEmpty(borderRadiusTokens) ? (
        <p className="label my-0">No border radius tokens</p>
      ) : (
        <div className="d-flex flex-row flex-wrap">
          {borderRadiusTokens.map((token: Token) => (
            <BorderRadiusToken
              key={token.id}
              token={token}
              onUpdate={() => {}}
              onDelete={() => {}}
            />
          ))}
        </div>
      )}
    </section>
  );
};

export default BorderRadiusTokens;
