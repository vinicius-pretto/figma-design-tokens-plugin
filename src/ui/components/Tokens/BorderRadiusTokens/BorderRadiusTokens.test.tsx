import * as React from "react";
import { render } from "../../../../testUtils";
import BorderRadiusTokens from "./BorderRadiusTokens";
import tokens from "../../../../testData/tokens";

describe("BorderRadiusTokens", () => {
  describe("snapshot", () => {
    it("no color tokens", () => {
      const { asFragment } = render(
        <BorderRadiusTokens tokens={[]} onDelete={jest.fn()} />
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it("with color tokens", () => {
      const { asFragment } = render(
        <BorderRadiusTokens tokens={tokens} onDelete={jest.fn()} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
