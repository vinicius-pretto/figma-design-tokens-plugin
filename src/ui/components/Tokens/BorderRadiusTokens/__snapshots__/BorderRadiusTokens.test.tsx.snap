// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderRadiusTokens snapshot no color tokens 1`] = `
<DocumentFragment>
  <section
    class="border-bottom p-sm"
  >
    <div
      class="tokens-section-header"
    >
      <h3
        class="section-title"
      >
        Border Radius
      </h3>
      <div
        aria-label="Add token"
        class="icon-button"
        data-testid="add-token-btn"
        role="button"
        tabindex="0"
      >
        <div
          class="icon icon--plus"
        />
      </div>
    </div>
    <p
      class="label my-0"
    >
      No border radius tokens
    </p>
  </section>
</DocumentFragment>
`;

exports[`BorderRadiusTokens snapshot with color tokens 1`] = `
<DocumentFragment>
  <section
    class="border-bottom p-sm"
  >
    <div
      class="tokens-section-header"
    >
      <h3
        class="section-title"
      >
        Border Radius
      </h3>
      <div
        aria-label="Add token"
        class="icon-button"
        data-testid="add-token-btn"
        role="button"
        tabindex="0"
      >
        <div
          class="icon icon--plus"
        />
      </div>
    </div>
    <div
      class="d-flex flex-row flex-wrap"
    >
      <div
        class="d-flex flex-row align-items-center justify-content-between w-100 color-token-container"
      >
        <button
          class="color-token"
        >
          <span
            class="mr-md border-radius-shape"
            style="border-radius: 4px;"
          >
            4
          </span>
          <span>
            border-radius-base
          </span>
        </button>
        <div
          class="d-flex flex-row"
        >
          <div
            aria-label="Edit border-radius-base"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--adjust"
            />
          </div>
          <div
            aria-label="Delete border-radius-base"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--trash"
            />
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-row align-items-center justify-content-between w-100 color-token-container"
      >
        <button
          class="color-token"
        >
          <span
            class="mr-md border-radius-shape"
            style="border-radius: 50px;"
          >
            50
          </span>
          <span>
            border-radius-circle
          </span>
        </button>
        <div
          class="d-flex flex-row"
        >
          <div
            aria-label="Edit border-radius-circle"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--adjust"
            />
          </div>
          <div
            aria-label="Delete border-radius-circle"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--trash"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
